config/prow/deploy.sh

```diff
@@ -124,14 +124,14 @@ kubectl config use-context gardener-prow-trusted
 kubectl apply --server-side=true -f "$SCRIPT_DIR/cluster/prowjob_customresourcedefinition.yaml"
 
 for c in "${prow_components[@]}"; do
-  kubectl apply --server-side=true -f "$SCRIPT_DIR/cluster/$c"
+  kubectl apply -f "$SCRIPT_DIR/cluster/$c"
```

Comment:
Why?

Reply:
When I changed it and ran the script manually prow self deployment [started to fail](https://prow.gardener.cloud/?job=post-ci-infra-deploy-prow). Apparently `deployer` service account does not have all permissions it needs to deploy prow completely. 
Afaik server-side deployment removes the last-applied-version annotations. Thus, deployer has to change every object when switching the server-side parameter and the lack of permissions becomes visible.
As it is related to roles and permissions I prefer to investigate changes in a separate pull request.

Reply:
Alright. Then let's do the switch to SSA later on.